virttest/qemu_devices/qcontainer.py

```diff
@@ -1433,125 +1412,81 @@ def machine_riscv64_mmio(cmd=False):
+                                             aobject="virtio-mmio-bus"))
             return devices
 
-        def machine_other(cmd=False):
+        def machine_other(machine_params):
             """
             isapc or unknown machine type. This type doesn't add any default
             buses or devices, only sets the cmdline.
-            :param cmd: If set uses "-M $cmd" to force this machine type
-            :return: List of added devices (including default buses)
             """
             LOG.warn('Machine type isa/unknown is not supported by '
                      'avocado-vt. False errors might occur')
-            devices = []
-            devices.append(qdevices.QStringDevice('machine', cmdline=cmd))
+            devices = [qdevices.QMachine(params=machine_params)]
             return devices
 
-        def secure_guest_handler(sectype, cmd):
-            """
-            Update machine option with secure guest information, e.g.
-              -machine q35,confidential-guest-support=lsec0
-            Note the secure guest object should have been created.
-            """
-            sev_mach_props = {'confidential-guest-support': '{obj_id}'}
-            tdx_mach_props = {'confidential-guest-support': '{obj_id}',
-                              'kvm-type': 'tdx'}
-            backend_props = {
-                'sev': ('sev-guest', sev_mach_props),
-                'tdx': ('tdx-guest', tdx_mach_props)
-            }
-
-            b, p = backend_props[sectype]
-            obj_id = self.get_by_params({"backend": b})[0].get_param('id')
-            sec_cmd = ','.join(
-                ['%s=%s' % (k, v) for k, v in p.items()]
-            ).format(obj_id=obj_id)
-
-            return '%s,%s' % (cmd, sec_cmd)
-
+        machine_params = {}
         machine_type = params.get('machine_type')
         machine_accel = params.get('vm_accelerator')
-        machine_type_extra_params = params.get('machine_type_extra_params')
-        if machine_type:
-            split_machine_type = machine_type.split(':', 1)
-            if len(split_machine_type) == 1:
-                avocado_machine = ''
+        machine_type_extra_params = params.get('machine_type_extra_params', '')
+        if machine_accel:
+            machine_params["accel"] = machine_accel
+        for keypair_str in machine_type_extra_params.split(','):
+            if not keypair_str:
+                continue
+            keypair = keypair_str.split("=", 1)
+            if len(keypair) < 2:
+                keypair.append("NO_EQUAL_STRING")
+            machine_params[keypair[0]] = keypair[1]
+
+        avocado_machine = ''
+        invalid_machine = None
+        if not machine_type:
+            for m_name, m_desc in self.__machines_info.items():
+                if '(default)' in m_desc:
+                    machine_type = m_name
+                    break
             else:
+                if machine_type is None:
+                    machine_type = "<unspecified>"
```

Comment:
```suggestion
                machine_type = "<unspecified>"
```

Reply:
Okay, the reason of adding this check is that we may want to distinguish empty strings as the value from the case by not given the param at all, to give users the right hint at the following warning message. We could discuss if needed, thanks!

Reply:
> Okay, the reason of adding this check is that we may want to distinguish empty strings as the value from the case by not given the param at all, to give users the right hint at the following warning message. We could discuss if needed, thanks!

All right, It looks good to me.
Thanks.